packages/salesforcedx-apex-replay-debugger/src/core/heapDumpService.ts

```diff
@@ -597,24 +630,44 @@ export class HeapDumpService {
   public createVariableFromReference(
     varName: string,
     refVariable: ApexVariableContainer,
-    visitedMap: Map<string, null>
+    visitedMap: Map<string, ApexVariableContainer | null>,
+    updateAfterVarCreation: ApexVariableContainer[]
```

Comment:
UpdateAfterVarCreation is used in the finally clause. I ended up getting into a chicken and egg situation here and some variable values cannot be updated until the variable is finished being created. This is only applicable to circular references. createVariableFromReference is a recursive function and because of the visitedMap we when we remove the last entry we know we're done with the initial variable being created. It's at this that time it's safe to update the values of any variables that we couldn't update during creation because we know the variable is complete. Trying to create the value earlier, when the variable was incomplete, would end up creating a value that was also incomplete.